File: crates/klyron_adapter/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron::_adapter {

AdapterBuilder::AdapterBuilder() : config_{} {}

AdapterBuilder& AdapterBuilder::set_config(const AdapterConfig& cfg) {
    config_ = cfg;
    return *this;
}

AdapterConfig AdapterBuilder::config() const {
    return config_;
}

}

File: crates/klyron_adapter/bindings/rust/builder.rs
//! Builder for klyron_adapter
use crate::types::AdapterConfig;

pub struct AdapterBuilder {
    config: AdapterConfig,
}

impl AdapterBuilder {
    pub fn new() -> Self {
        Self { config: AdapterConfig::default() }
    }
    pub fn with_version(mut self, version: &str) -> Self {
        self.config.version = version.to_string();
        self
    }
    pub fn build(self) -> AdapterConfig {
        self.config
    }
}
